app/cdap/components/hydrator/helpers/DAGhelpers.tsx

```diff
@@ -0,0 +1,404 @@
+    rename: 'icon-rename',
+    archive: 'icon-archive',
+    wrangler: 'icon-DataPreparation',
+    normalize: 'icon-normalize',
+    xmlmultiparser: 'icon-XMLmultiparser',
+    xmltojson: 'icon-XMLtoJSON',
+    decisiontreepredictor: 'icon-decisiontreeanalytics',
+    decisiontreetrainer: 'icon-DesicionTree',
+    hashingtffeaturegenerator: 'icon-HashingTF',
+    ngramtransform: 'icon-NGram',
+    tokenizer: 'icon-tokenizeranalytics',
+    skipgramfeaturegenerator: 'icon-skipgram',
+    skipgramtrainer: 'icon-skipgramtrainer',
+    logisticregressionclassifier: 'icon-logisticregressionanalytics',
+    logisticregressiontrainer: 'icon-LogisticRegressionclassifier',
+    hdfsdelete: 'icon-hdfsdelete',
+    hdfsmove: 'icon-hdfsmove',
+    windowssharecopy: 'icon-windowssharecopy',
+    httppoller: 'icon-httppoller',
+    window: 'icon-window',
+    run: 'icon-Run',
+    oracleexport: 'icon-OracleDump',
+    snapshottext: 'icon-SnapshotTextSink',
+    errorcollector: 'fa-exclamation-triangle',
+    mainframereader: 'icon-MainframeReader',
+    fastfilter: 'icon-fastfilter',
+    trash: 'icon-TrashSink',
+    staterestore: 'icon-Staterestore',
+    topn: 'icon-TopN',
+    wordcount: 'icon-WordCount',
+    datetransform: 'icon-DateTransform',
+    sftpcopy: 'icon-FTPcopy',
+    sftpdelete: 'icon-FTPdelete',
+    validatingxmlconverter: 'icon-XMLvalidator',
+    wholefilereader: 'icon-Filereader',
+    xmlschemaaction: 'icon-XMLschemagenerator',
+    s3toredshift: 'icon-S3toredshift',
+    redshifttos3: 'icon-redshifttoS3',
+    verticabulkexportaction: 'icon-Verticabulkexport',
+    verticabulkimportaction: 'icon-Verticabulkload',
+    loadtosnowflake: 'icon-snowflake',
+    kudu: 'icon-apachekudu',
+    orientdb: 'icon-OrientDB',
+    recordsplitter: 'icon-recordsplitter',
+    scalasparkprogram: 'icon-spark',
+    scalasparkcompute: 'icon-spark',
+    cdcdatabase: 'icon-database',
+    cdchbase: 'icon-hbase',
+    cdckudu: 'icon-apachekudu',
+    changetrackingsqlserver: 'icon-database',
+    conditional: 'fa-question-circle-o',
+  };
+
+  const pluginName = plugin ? plugin.toLowerCase() : '';
+  const icon = iconMap[pluginName] ? iconMap[pluginName] : 'fa-plug';
+  return icon;
+}
+
+export function getNodesMap(nodes) {
+  const nodesMap = {};
+  nodes.forEach((node) => {
+    nodesMap[node.name] = node;
+  });
+  return nodesMap;
+}
+
+function customGraphLayout(graph, nodes, connections) {
+  const graphNodes = graph._nodes;
+  const nodesMap = getNodesMap(nodes);
+
+  nodes.forEach((node) => {
+    const location = graphNodes[node.name];
+    let locationX = location.x;
+    let locationY = location.y;
+
+    if (node.type === 'alertpublisher' || node.type === 'errortransform') {
+      const connToThisNode = connections.find((conn) => conn.to === node.name);
+      if (connToThisNode) {
+        const sourceNode = connToThisNode.from;
+        let nonErrorsAlertsConnCount = 0;
+        for (let i = 0; i < connections.length; i++) {
```

Comment:
node is giving me this: `prefer-for-of: Expected a 'for-of' loop instead of a 'for' loop with this simple iteration`

Reply:
i see that too - i don't know where that rule is coming from but i disagree with it. There is nothing wrong with using a plain for loop... will change the tslint